### DISCUSSION TITLE
Allow to disable global usings per file

### DISCUSSION THREAD

**Discussion by Dreamescaper** [dotnet/csharplang#7164] [Language Ideas]
I'm having a small issue with files generated by a source generator. The problem is that in some case global usings cause the ambiguous references, when SG is not aware of  them.
Therefore, would be nice to be able to disable global usings per file basis.

Maybe with some directive, like
```
#global-usings disabled
```

**Comment by colejohnson66** [comment]
Your source generator can work around this by using `global::` syntax for type names.

**Reply by Dreamescaper** [reply]
Sure, that's what I do now.
But the readability is not great (that's not mega critical for generated files, but I still try to make them readable).
